thoth/lab/adviser.py

```diff
@@ -0,0 +1,322 @@
+        hovertext=[y[0] for y in justifications_df[["message"]].values],
+        hoverinfo="text",
+        marker=dict(
+            color=justifications_df["count"], colorscale="Viridis", opacity=0.8, showscale=True  # choose a colorscale
+        ),
+    )
+
+    data = [trace1]
+
+    margin = {
+        "l": 0,
+        "r": 0,
+        "b": 0,
+        "t": 0,
+    }
+
+    layout = go.Layout(
+        title="Adviser justifications",
+        margin=margin,
+        scene=dict(xaxis=dict(title="Justification encoded ID"), yaxis=dict(title="Counter")),
+        showlegend=True,
+        legend=dict(orientation="h"),
+    )
+
+    fig = go.Figure(data=data, layout=layout)
+
+    iplot(fig, filename="bar-plot")
+
+    return justifications_df
+
+
+def _aggregate_data_per_interval(adviser_justification_df: pd.DataFrame, intervals: int = 10):
+    """Aggregate advise justifications per time intervals."""
+    begin = min(adviser_justification_df["date"].values)
+    end = max(adviser_justification_df["date"].values)
+    timestamps = []
+    delta = (end - begin) / intervals
+    value = begin
+    for i in range(1, intervals + 1):
+        value = value + delta
+        timestamps.append(value)
+    timestamps[0] = begin
+    timestamps[len(timestamps) - 1] = end
+
+    aggregated_data = {}
+    for l in range(0, len(timestamps)):
+        low = timestamps[l - 1]
+        high = timestamps[l]
+        aggregated_data[high] = {}
+        subset_df = adviser_justification_df[
+            (adviser_justification_df["date"] >= low) & (adviser_justification_df["date"] <= high)
+        ]
+
+        for index, row in subset_df[["jm_hash_id_encoded", "message", "date"]].iterrows():
+            encoded_id = row["jm_hash_id_encoded"]
+            if encoded_id not in aggregated_data[high].keys():
+                aggregated_data[high][encoded_id] = {}
+                aggregated_data[high][encoded_id]["jm_hash_id_encoded"] = f"type-{encoded_id}"
+                aggregated_data[high][encoded_id]["message"] = row["message"]
+                aggregated_data[high][encoded_id]["count"] = subset_df["jm_hash_id_encoded"].value_counts()[encoded_id]
+
+    return aggregated_data
+
+
+def _create_heatmaps_values(input_data: dict, advise_encoded_type: List[int]):
+    """Create values for heatmaps.
+
+    :param adviser_justification_df: data frame as returned by `create_final_dataframe' per identifier.
+    """
+    heatmaps_values = []
+    for t in set(advise_encoded_type):
+        type_values = []
+        for interval_runs in input_data.values():
+            if t in interval_runs.keys():
+                type_values.append(interval_runs[t]["count"])
+            else:
+                type_values.append(0)
+
+        heatmaps_values.append(type_values)
+
+    return heatmaps_values
+
+
+def create_adviser_heatmap(
+    adviser_justification_df: pd.DataFrame, save_result: bool = False, project_folder: str = "", folder_name: str = ""
```

Comment:
Is there any reason for having `project_folder` and `folder_name` instead of one `output_dir` parameter? Also, I might consider renaming to `output_*` instead of `project_folder`, it might be a bit confusing what this parameter is in fact for.

Reply:
> Is there any reason for having `project_folder` and `folder_name` instead of one `output_dir` parameter? Also, I might consider renaming to `output_*` instead of `project_folder`, it might be a bit confusing what this parameter is in fact for.

It's to split the different type of plot in case many are created, actually, this logic is equal also in other plot functions created, I can generalize it.